cty/function/stdlib/collection.go

```diff
@@ -121,7 +122,9 @@ var LengthFunc = function.New(&function.Spec{
 		return cty.Number, nil
 	},
 	Impl: func(args []cty.Value, retType cty.Type) (ret cty.Value, err error) {
-		return args[0].Length(), nil
+		// remove and re-apply top-level marks only
+		collection, marks := args[0].Unmark()
+		return collection.Length().WithMarks(marks), nil
```

Comment:
In this particular case, `Length()` here is an operation method rather than an integration method (that is, it stays within the space of cty values rather than exporting into "normal" Go types) and so it already handles marks internally itself, so I think it would be safe to just retain the old body here and just add the `AllowMarked: true` annotation above, to get the same result.

Reply:
Good catch, thanks!